cbw_api_toolbox/cbw_api.py

```diff
@@ -31,32 +31,39 @@ def __init__(self, api_url, api_key, secret_key, verify_ssl=False):
     def _get_params(params=None):
         return "/{0}".format('/'.join(params)) if params else ""
 
-    def _request(self, route, params=None):
+    def _build_route(self, route, params=None):
+        return "{0}{1}{2}".format(self.api_url, route, self._get_params(params))
+
+    def _request(self, verb, route, body_params=None):
         response = None
+
+        if body_params is not None:
+            body_params = json.dumps(body_params)
         try:
-            response = requests.get("{0}{1}{2}".format(self.api_url, route,
-                                                       self._get_params(params)),
-                                    auth=CBWAuth(self.api_key, self.secret_key),
-                                    verify=self.verify_ssl)
+            response = requests.request(
+                verb,
+                route,
+                data=body_params,
+                auth=CBWAuth(self.api_key, self.secret_key),
+                verify=self.verify_ssl)
             return json.loads(response.text)
 
         except JSONDecodeError as error:
-            self.logger.error("An error occurred when decoding {0} with route {1}/{2}{3} : {4}"
-                              .format(response.text, self.api_url, route,
-                                      self._get_params(params), error))
+            self.logger.error("An error occurred when decoding %s with route %s : %s",
```

Comment:
Please use .format

Reply:
Pylint doesn't let me use format (Pylint message: "Use % formatting in logging functions and pass the % parameters as arguments")